app/models/instance.rb

```diff
@@ -29,6 +30,14 @@ class Instance < ApplicationRecord
   scope :by_domain_and_subdomains, ->(domain) { where("reverse('.' || domain) LIKE reverse(?)", "%.#{domain}") }
   scope :with_domain_follows, ->(domains) { where(domain: domains).where(domain_account_follows) }
 
+  def self.find_or_initialize_by_domain(domain)
+    normalized_domain = TagManager.instance.normalize_domain(domain&.strip)
+
+    raise ActiveRecord::RecordInvalid if normalized_domain.blank?
```

Comment:
I don't think this is really the right error to raise here, open to suggestions. Would it be worth adding something more specific to `lib/exceptions.rb` ? e.g., `Mastodon::InvalidDomainError` or something?

This should probably also raise if the normalized domain is invalid, though it looks like `Addressable::URI::InvalidURIError` will be raised from `TagManager.instance.normalize_domain` in that case?

Reply:
oddly enough, `TagManager.instance.normalize_domain` doesn't raise if the domain is blank or otherwise malformed..

So these are currently "valid" domains:
- `''` (returns an empty string)
- `.` (returns `.`)
- `a.` (returns `a`)
- `.a` (returns `.a`)